apps/render_cf.py

```diff
@@ -158,6 +164,45 @@ def render_templates(job_types, security_environment, api_name):
         template_file.with_suffix('').write_text(output)
 
 
+def get_compute_environments(job_types: dict, compute_env_file: Path) -> list[dict]:
+    compute_envs = []
+    compute_env_names = []
+    compute_env_imports = set()
+    for _, job_spec in job_types.items():
+        for task in job_spec['tasks']:
+            compute_env = task['compute_environment']
+            if 'name' in compute_env:
+                name = compute_env['name']
+                if name in compute_env_names:
+                    raise NameError(
+                        f'Compute envs must have unique names but the following is defined more than once: {name}.'
+                    )
+                compute_envs.append(compute_env)
+                compute_env_names.append(name)
+            elif 'import' in compute_env and compute_env['import'] != 'Default':
+                compute_env_imports.add(compute_env['import'])
+
+    if compute_env_file:
+        compute_envs_from_file = yaml.safe_load(compute_env_file.read_text())['compute_environments']
+        for name in compute_envs_from_file:
+            if name in compute_env_names:
+                raise NameError(
```

Comment:
See previous comment.
```suggestion
                raise ValueError(
```